client/src/components/PortfolioTable.tsx

```diff
@@ -0,0 +1,741 @@
+                                ? cell.column.getSize() || undefined
+                                : undefined,
+                          }}
+                        >
+                          {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                        </td>
+                      );
+                    })}
+                  </tr>
+                  {row.getIsExpanded() && (
+                    <tr>
+                      {/* 2nd row is a custom 1 cell row */}
+                      <td colSpan={row.getVisibleCells().length}>{renderContacts({ row })}</td>
+                    </tr>
+                  )}
+                </Fragment>
+              );
+            })}
+          </tbody>
+        </table>
+        <div className="pagination">
+          <div className="prev">
+            <button
+              className="page-btn"
+              onClick={() => table.previousPage()}
+              disabled={!table.getCanPreviousPage()}
+            >
+              {i18n._(t`Previous`)}
+            </button>
+          </div>
+          <div className="center">
+            <span className="page-info">
+              <span>
+                <Trans>Page</Trans>
+              </span>
+              <div>
+                <input
+                  type="number"
+                  value={String(table.getState().pagination.pageIndex + 1)}
+                  onChange={(e) => {
+                    const page = e.target.value ? Number(e.target.value) - 1 : 0;
+                    table.setPageIndex(page);
+                  }}
+                />
+              </div>
+              <Trans>of</Trans> <span className="total-pages">{table.getPageCount()}</span>
+            </span>
+            <select
+              value={table.getState().pagination.pageSize}
+              onChange={(e) => {
+                table.setPageSize(Number(e.target.value));
+              }}
+            >
+              {[10, 20, 50, 100, 500].map((pageSize) => (
+                <option key={pageSize} value={pageSize}>
+                  Show {pageSize}
+                </option>
+              ))}
+            </select>
+          </div>
+          <div className="next">
+            <button
+              className="page-btn"
+              onClick={() => table.nextPage()}
+              disabled={!table.getCanNextPage()}
+            >
+              {i18n._(t`Next`)}
+            </button>
+          </div>
+        </div>
+      </div>
+    );
+  })
+);
+
+const renderContacts = ({ row }: { row: Row<AddressRecord> }) => {
+  return (
+    <ul className="contacts-list">
+      {Object.entries(_groupBy(row.original.allcontacts, "value"))
+        .sort(sortContactsByImportance)
+        .map((group) => group[1][0])
+        .map((contact, i) => (
+          <li key={i}>
+            {contact.title}: {contact.value}
+          </li>
+        ))}
+    </ul>
+  );
+};
+
+type ArrowIconProps = {
```

Comment:
could be consolidated with `Icons.tsx` maybe

Reply:
oh yeah good idea!